processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1474,7 +1474,15 @@ public Rowboat apply(Rowboat lhs, Rowboat rhs)
       Object[] rhsMetrics = rhs.getMetrics();
 
       for (int i = 0; i < metrics.length; ++i) {
-        metrics[i] = metricAggs[i].combine(lhsMetrics[i], rhsMetrics[i]);
+        Object lhsMetric = lhsMetrics[i];
+        Object rhsMetric = rhsMetrics[i];
+        if (lhsMetric == null) {
+          metrics[i] = rhsMetric;
+        } else if (rhsMetric == null) {
+          metrics[i] = lhsMetric;
```

Comment:
Anyway, I think conversion/interpretation of null is probably best left to the serializer and aggregator, that comment regarding types was in reference to this comment in #2744:

https://github.com/druid-io/druid/issues/2744#issuecomment-203144075